python/mxnet/gluon/loss.py

```diff
@@ -696,3 +696,35 @@ def hybrid_forward(self, F, pred, positive, negative):
                      axis=self._batch_axis, exclude=True)
         loss = F.relu(loss + self._margin)
         return _apply_weighting(F, loss, self._weight, None)
+
+
+class CenterLoss(Loss):
+    r""" Center Loss
+    """
+    def __init__(self, num_classes, feature_size, ctx, alpha=0.5, lmbd=0.01, weight=1., batch_axis=0, **kwargs):
+        super(CenterLoss, self).__init__(weight*lmbd, batch_axis, **kwargs)
+        self._alpha = alpha
+        self._lmbd = lmbd
+        self._num_classes = num_classes
+        self._feature_size = feature_size
+        self._ctx = ctx
+        self._centers = nd.zeros((num_classes, feature_size), ctx=ctx)
+
+    def hybrid_forward(self, F, output, label, sample_weight=None):
+        selected_centers = F.take(self._centers, label)
+        diff = output - selected_centers
+        loss = F.sum(F.square(diff), axis=1)
+
+        loss = _apply_weighting(F, loss, self._weight, sample_weight)
+        # Update the centers
+        counts = nd.zeros((self._num_classes, ), ctx=self._ctx)
```

Comment:
using nd means this is not a hybridblock anymore

Reply:
Sorry, I did not get it. @szha suggested updating the centers manually. What should I use in this case?

Reply:
What are you referring to? Here if you explicitly use ndarray, the hybridized version or the symbolic version of training won't work properly, hence it's discouraged to use ndarray-specific functionality in HybridBlocks. Loss class is a child class of HybridBlock.

Reply:
So how can update the centers for each class? using the ```params```?

Reply:
Declaring a parameter is the usual way of keeping states in operators. I'm not sure about what way would be appropriate for the update though.